### ISSUE TITLE
UI ninepatch border geometry should match the borders in the layout (maybe)

### ISSUE THREAD

**Issue by ickshonpe** [bevyengine/bevy#14943]
## What problem does this solve or what need does it fill?

Borders created using the UI's  texture slicing api aren't borders in the UI layout.
This means that for example that left-aligned text will align at the left edge of the uinode, not the inner edge of the border, and write over the ninepatch border:

<img width="308" alt="nbh" src="https://github.com/user-attachments/assets/3f13427a-77be-48a9-8dab-800f7e402145">

## What solution would you like?

Have an option to draw borders from the UI layout using the ninepatch api.

## What alternative(s) have you considered?

You can add an invisable border the same size as the ninepatch border. This might be awkward to keep synchronised though.

**Comment by alice-i-cecile** [comment]
Yeah I *think* that this should end up creating borders? I don't hold that opinion very strongly though.

**Comment by ickshonpe** [comment]
> Yeah I _think_ that this should end up creating borders? I don't hold that opinion very strongly though.

Yeah I'm not very sure about how this should work. I've reimplemented the UI texture slicing stuff to use a shader and I was trying to match my changes to the old api but it feels like it would actually be easier just to use the border geometry from the layout because I'm not sure how the `min_corner_scale` field is meant to work :sweat:.

**Comment by ickshonpe** [comment]
Another problem is that the nine-patcher doesn't just draw a border but also a background in the center. We could just not draw the center patch though, depending.

**Comment by ickshonpe** [comment]
Thought about this and I think all it needs is a `UiImageScaleMode` component copying `ImageScaleMode` but without the `max_corner_size` field, and it will work fine.
